ioctl/cmd/znode/znodemessagequery.go

```diff
@@ -0,0 +1,84 @@
+package znode
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/output"
+)
+
+var (
+	// znodeMessageQuery represents the znode query message command
+	znodeMessageQuery = &cobra.Command{
+		Use:   "query",
+		Short: config.TranslateInLang(znodeMessageQueryShorts, config.UILanguage),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			id, err := cmd.Flags().GetString("message-id")
+			if err != nil {
+				return errors.Wrap(err, "failed to get flag message-id")
+			}
+
+			out, err := queryMessageStatus(id)
+			if err != nil {
+				return output.PrintError(err)
+			}
+			output.PrintResult(out)
+			return nil
+		},
+	}
+
+	// znodeMessageSendShorts znode message query shorts multi-lang support
+	znodeMessageQueryShorts = map[config.Language]string{
+		config.English: "query message status from znode",
+		config.Chinese: "向znode查询消息状态",
+	}
+
+	_flagMessageIDUsages = map[config.Language]string{
+		config.English: "message id",
+		config.Chinese: "消息ID",
+	}
+)
+
+func init() {
+	znodeMessageQuery.Flags().StringP("message-id", "i", "", config.TranslateInLang(_flagMessageIDUsages, config.UILanguage))
+	_ = znodeMessageQuery.MarkFlagRequired("message-id")
+}
+
+func queryMessageStatus(id string) (string, error) {
+	u := url.URL{
+		Scheme: "http",
+		Host:   config.ReadConfig.ZnodeEndpoint,
+		Path:   "/message/" + id,
+	}
+
+	rsp, err := http.Get(u.String())
```

Comment:
suggest to use more advanced libraries for http requests to make it simple, like resty.